StripeConnect/StripeConnect/Source/Internal/FinancialConnectionsPresenter.swift

```diff
@@ -9,21 +9,35 @@
 import StripeFinancialConnections
 import UIKit
 
+protocol FinancialConnectionSheetImplementation: AnyObject {
+    init(financialConnectionsSessionClientSecret: String, returnURL: String?)
+    var apiClient: STPAPIClient { get set}
+    func presentForToken(
+        from presentingViewController: UIViewController,
+        completion: @escaping (FinancialConnectionsSheet.TokenResult) -> Void
+    )
+}
+
+extension FinancialConnectionsSheet: FinancialConnectionSheetImplementation {}
+
 /// Wraps `FinancialConnectionsSheet` for easy dependency injection in tests
 class FinancialConnectionsPresenter {
     @MainActor
+    @available(iOS 15, *)
     func presentForToken(
-        apiClient: STPAPIClient,
+        componentManager: EmbeddedComponentManager,
         clientSecret: String,
         connectedAccountId: String,
-        from presentingViewController: UIViewController
+        from presentingViewController: UIViewController,
+        financialConnectionSheetInitializer: (_ secret: String, _ returnURL: String?) -> FinancialConnectionSheetImplementation = FinancialConnectionsSheet.init
     ) async -> FinancialConnectionsSheet.TokenResult {
-        let financialConnectionsSheet = FinancialConnectionsSheet(
-            financialConnectionsSessionClientSecret: clientSecret
-        )
+        let financialConnectionsSheet = financialConnectionSheetInitializer(clientSecret, nil)
         // FC needs the connected account ID to be configured on the API Client
         // Make a copy before modifying so we don't unexpectedly modify the shared API client
-        financialConnectionsSheet.apiClient = apiClient.makeCopy()
+        financialConnectionsSheet.apiClient = componentManager.apiClient.makeCopy()
+        
+        // FC expects a public key and not a UK. If there is a public key override we should use that.
+        financialConnectionsSheet.apiClient.publishableKey = componentManager.publicKeyOverride ?? financialConnectionsSheet.apiClient.publishableKey
```

Comment:
nit: I'd rather keep the logic for copying the PK encapsulated in `makeCopy()`, in the event that ever needs to change. So instead of defaulting this value to `financialConnectionsSheet.apiClient.publishableKey`, can we instead only override it if there is a PK override?
```suggestion
        if let publicKeyOverride = componentManager.publicKeyOverride {
            financialConnectionsSheet.apiClient.publishableKey = publicKeyOverride
        }
```